### PR TITLE
Add CDK for ArgoCD and Prow

### PR DESCRIPTION
**Pull request by RedbackThomson**
Please see the documentation for the new infrastructure in `docs/infrastructure.md`

- Adds the test infrastructure CDK template in `infra/`
- Adds the ArgoCD applications in `argo/`
- Adds configuration for Prow and its jobs in `prow/`

Still left to do:
- Properly configure the log bucket and MinIO permissions
- Connect the public domain name with Route53

### CODE REVIEW

File: argo/prow-config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: prow-config
+  namespace: argocd
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/RedbackThomson/ack-test-infra.git
```
Comment (jaypipes) [inline argo/prow-config.yaml:9]: oops :)

File: argo/prow-jobs.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: prow-jobs
+  namespace: argocd
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/RedbackThomson/ack-test-infra.git
```
Comment (jaypipes) [inline argo/prow-jobs.yaml:9]: ditto :)

File: argo/prow-config.yaml (hunk 1 above)
Comment (RedbackThomson) [reply argo/prow-config.yaml:9]: Not quite oops. Just for faster iteration. Very much on purpose, will submit a fix up before merge

Comment (jaypipes) [reply argo/prow-config.yaml:9]: I'd be good to merge this as-is if the above pointed to the main repo. :)

**Review by vijtrip2 — APPROVED** [review]
LGTM

**Review by jaypipes — APPROVED** [review]

**Pull request merged**